dbcontrollers/products.go

```diff
@@ -148,7 +176,12 @@ func (*MYSQLController) GetProduct(productID *uuid.UUID) (*models.ProductData, e
 
 	product, err := mysqldb.Functions.GetProductByID(productID, tx)
 	if err != nil {
-		return nil, err
+		if err == sql.ErrNoRows {
+			if err := mysqldb.DBConnector.Rollback(tx); err != nil {
```

Comment:
In case of any update/insert/delete yes, in order to undo the changes happening during the failure. In case of selects it won't do harm if there is no rollback.

But since I have multiple queries in one transaction, mixed selects/updates/deletes/inserts I would like to keep consistency and make sure any failed query will rollback the whole transaction.
I am not experienced enough to tell if this is the optimal way, but it certainly will not do any harm having a bit over engineered safety in this case.

Reply:
Ah alright. I completely missread your question.:D The answer above is correct but not answer your question. So there are cases, when the expectation is return noRows error. So, it is the success path. This means that in the low level sql.ErrNoRows always returns without rollback. It is up to the dbcontrollers function to determine if this error is a fail case or the expected case. It will rollback if the error represents a failure and will not roll back if that is what is the expectation.
For example:
You want to add a new user sql.ErrNoRows will mean that yes, there is no new user the code can progress. But if you want to get certain user this error means that the code failed to find any user stop what it was doing and roll back everything.

Reply:
As I am writing, I feel this maybe no entirely the best way to approach. If you think it can be misleading, please raise a task to optimize it in the future.